PSReadLine/DynamicHelp.cs

```diff
@@ -0,0 +1,289 @@
+
+                var bufferWidth = console.BufferWidth;
+                var columnWidth = this.ColumnWidth;
+
+                var items = this.ItemsToDisplay;
+                for (var row = 0; row < this.Rows; row++)
+                {
+                    var cells = 0;
+                    if (row >= items.Count)
+                    {
+                        break;
+                    }
+
+                    var itemToWrite = items[row];
+
+                    if(itemToWrite.Length > bufferWidth)
+                    {
+                        multilineItems++;
+                    }
+
+                    console.Write(itemToWrite);
+                    cells += columnWidth;
+
+                    // Make sure we always write out exactly 1 buffer width to erase anything
+                    // from a previous menu.
+                    if (cells < bufferWidth)
+                    {
+                        // 'BlankRestOfLine' erases rest of the current line, but the cursor is not moved.
+                        console.BlankRestOfLine();
+                    }
+
+                    // Explicit newline so consoles see each row as distinct lines, but skip the
+                    // last line so we don't scroll.
+                    if (row != (this.Rows - 1))
+                    {
+                        AdjustForPossibleScroll(1);
+                        MoveCursorDown(1);
+                    }
+                }
+
+                if (multilineBlock != null)
+                {
+                    if (Rows < multilineBlock.Rows)
+                    {
+                        // If the last menu row took the whole buffer width, then the cursor could be pushed to the
+                        // beginning of the next line in the legacy console host (NOT in modern terminals such as
+                        // Windows Terminal, VSCode Terminal, or virtual-terminal-enabled console host). In such a
+                        // case, there is no need to move the cursor to the next line.
+                        //
+                        // If that is not the case, namely 'CursorLeft != 0', then the rest of the last menu row was
+                        // erased, but the cursor was not moved to the next line, so we will move the cursor.
+                        if (console.CursorLeft != 0)
+                        {
+                            // There are lines from the previous rendering that need to be cleared,
+                            // so we are sure there is no need to scroll.
+                            MoveCursorDown(1);
+                        }
+
+                        Singleton.WriteBlankLines(multilineBlock.Rows - Rows);
+                    }
+                }
+            }
+
+            public void Clear()
+            {
+                // Add 1 for the movement to next line before displaying block
+                _singleton.WriteBlankLines(Top, Rows + multilineItems + 1);
```

Comment:
Can you please explain a little more about why `+ 1` here?

Reply:
Not needed anymore. The help content I was testing with had a \r\n. I trim that now.